jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java

```diff
@@ -144,22 +199,28 @@ public void sample(final SampleResult result, final boolean isUseSenderTime) {
     }
 
     private void messageReceived(final Message message) {
-        try {
-            synchronized (lock) {
-                final int bodyLength = getValue(message.getBody()).length;
+        synchronized (lock) {
+            messageCount++;
+            LOGGER.trace("Received message. count : {}", messageCount);
+            byte[] messageBody = getValue(message.getBody());
+            messageSize += messageBody.length;
+    
+            if (sampler.isUseSenderTime()) {
+                sampleEnd = System.currentTimeMillis();
+                LOGGER.debug("Message received time : {}", sampleEnd);
+                if (sampler.isSenderTimeInPayload()) {
+                    final Long time = (long) getJSONValue(message.getBody())
+                            .get(sampler.getSenderTimeVariableName());
```

Comment:
If getJSONValue gives a empty map the get will give null as return value.

Reply:
Yes I will return in `null `value. Thats why the method `verifySenderTimeAndSetSamplingTime` checks for `null` value.